specs/sharding/beacon-chain.md

```diff
@@ -402,7 +402,7 @@ def get_shard_proposer_index(beacon_state: BeaconState, slot: Slot, shard: Shard
     """
     epoch = compute_epoch_at_slot(slot)
     committee = get_shard_committee(beacon_state, epoch, shard)
-    seed = hash(get_seed(beacon_state, epoch, DOMAIN_BEACON_PROPOSER) + uint_to_bytes(beacon_state.slot))
+    seed = hash(get_seed(beacon_state, epoch, DOMAIN_BEACON_PROPOSER) + uint_to_bytes(slot))
```

Comment:
We should probably assert that the slot is of the current epoch or older (and not older than randao seed history). The usage of this function in this doc is safe currently, but we may use it elsewhere. The phase0 beacon-proposer variant uses `state.slot` because it only computes for the current slot, and thus does not have the same safety problem.